defensics/defensics_runner.sh

```diff
@@ -0,0 +1,278 @@
+composed_att_sr_cmd=${composed_att_sr_cmd:1}
+
+# 4 .Bluetooth-LE-HOGP-Host
+LOCAL_NAME_HOGP="synopsys"
+IO_CAPABILITY_HOGP="no-input-no-output"
+# Require Man-in-te-middle protection using --require-mitm, --no-require-mitm otherwise
+REQUIRE_MITM="--no-require-mitm"
+# Require bonding protection using --require-bonding, --no-require-bonding otherwise
+REQUIRE_BONDING="--no-require-bonding"
+hogp_host_cmds=(
+  "--local-name $LOCAL_NAME_HOGP"
+  "--io-capability $IO_CAPABILITY_HOGP"
+  $REQUIRE_MITM
+  $REQUIRE_BONDING
+)
+composed_hogp_host_cmd=$(printf " %s" "${hogp_host_cmds[@]}")
+composed_hogp_host_cmd=${composed_hogp_host_cmd:1}
+
+# 5. Bluetooth-LE-SMP-Client
+LOCAL_NAME_SMP="synopsys"
+# If ACL link should be open between test cases choose --keep-link and --no-keep-link otherwise
+KEEP_LINK_SMP="--no-keep-link"
+# Enabling internal ATT server - if enabled, ATT requests will be responded to and ignored otherwise.
+# Use --enable-att-server or --disable-att-server
+ENABLE_ATT_SR_SMP="--enable-att-server"
+# UUID to advertise when in connectable mode, should be in hexadecimal format
+UUID="0x180a"
+# Request bonding - enables exchanged keys storing. Value --bonding or --no-bonding
+BONDING="--bonding"
+PASSKEY_SMP="000000"
+smp_cl_cmds=(
+  "--local-name $LOCAL_NAME_SMP"
+  $KEEP_LINK_SMP
+  $ENABLE_ATT_SR_SMP
+  "--uuid $UUID"
+  $BONDING
+  "--passkey $PASSKEY_SMP"
+)
+composed_smp_cl_cmd=$(printf " %s" "${smp_cl_cmds[@]}")
+composed_smp_cl_cmd=${composed_smp_cl_cmd:1}
+
+log_path=""
+log_path=$RESULTS_PATH$FOLDER_NAME$TEST_CODE
+
+cmds=(
+  "sudo -i java -jar $MONITOR_PATH"
+
+  # [options]
+  $selection_cmd
+  $trim_cmd
+  $VERBOSE
+  "--exec-after-test-run $AFTER_RUN"
+#  "--max-fails-limit 1"
+  # [suite]
+  "--suite $SUITE"
+  # [suite options]
+  "--log-dir $log_path"
+  "--no-loop"
+  "--exec-test-run $BEFORE_RUN"
+  "--exec-pre-test-case $BEFORE_CASE"
+  "--exec-instrument $AS_INSTRUMENTATION"
+  "--exec-post-test-case $AFTER_EACH"
+  "--exec-instrument-fail $INSTRUMENT_FAIL"
+  "$test_indicies_cmd"
+)
+
+# construct command and run defensics
+composed_cmd=$(printf " %s" "${cmds[@]}")
+composed_cmd=${composed_cmd:1}
+# append to command suite specific settings
+if [ "$suite_name" = "btle-ad-510" ] ; then
+  complete_cmd=$composed_cmd" "$composed_adv_cmd
+elif [ "$suite_name" = "btle-attc-530" ] ; then
+  complete_cmd=$composed_cmd" "$composed_att_cl_cmd
+elif [ "$suite_name" = "btle-att-520" ] ; then
+  complete_cmd=$composed_cmd" "$composed_att_sr_cmd
+elif [ "$suite_name" = "btle-hogp-510" ] ; then
+  complete_cmd=$composed_cmd" "$composed_hogp_host_cmd
+elif [ "$suite_name" = "btle-smpc-600" ] ; then
+  complete_cmd=$composed_cmd" "$composed_smp_cl_cmd
+# if other suite is used, no settings are added
+else
+  complete_cmd=$composed_cmd
+fi
+
+
+# launch main.py (server) in seperate terminal. This should be closed manually after run ends
+gnome-terminal --tab -e "python3 main.py"
```

Comment:
Not everyone is using gnome-terminal. I think we should just manually start the server in another terminal before running this script.